.github/workflows/main.yml

```diff
@@ -0,0 +1,79 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI Build
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  
+  linux:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+    - name: wget
+      uses: wei/wget@v1
+      with:
+        args: -O nuget.exe https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
```

Comment:
I think it's simpler if you just install wget with `sudo apt install -y wget` and just use it, instead of depending on an external action

Reply:
wget is installed by default but it gives HTTP 404 Error when you try to download anything

Reply:
how can it give 404? I use it all the time myself, maybe switch to curl?

Reply:
Take a look:
https://github.com/aarani/TLSharp/runs/551697235?check_suite_focus=true
https://github.com/aarani/TLSharp/runs/551693010?check_suite_focus=true

Reply:
did you add a comma at the end of the URL by mistake?

Reply:
Yep you're right i thought it's yaml syntax at first but it wasn't

Reply:
great, let's download nuget.exe manually then

Reply:
Done